README.md

```diff
@@ -44,8 +44,13 @@ java -jar zipkin-sparkstreaming-job.jar \
   --zipkin.sparkstreaming.sparkMaster=spark://127.0.0.1:7077
 ```
 
+
 ## Key Components
 
+The image below shows the internal architecture of zipkin spark streaming job. StreamFactory is a extensible interface that ingests data from Kafka or any other transport. The filtering step filters spans based on criteria like service name(#33). The aggregation phase groups the spans by time or trace ID. The adjuster phase is useful for making adjustments to spans that belong to the same trace. For example, the FinagleAdjuster fixes known bugs in the old finagle zipkin tracer. The final consumer stage persists the data to a storage system like ElasticSearch service. 
```

Comment:
issue links don't automatically work in markdown like this. you'd need to add a full url I think (instead of 33).